toqito/state_props/is_separable.py

```diff
@@ -161,6 +185,79 @@ def is_separable(state: np.ndarray, dim: None | int | list[int] = None, level: i
     #    Xt = state
     # commented out because pylint flagged this as an unused variable
 
+    # There are conditions that are both necessary and sufficient when both
```

Comment:
what's the reference for the lines after this? I want to read over why you used `p = np.zeros((6, 7, 8, 9))` and why the elements in the `np.array` for `F` are hard coded.

Reply:
It's taken from QETLAB. See [here](https://github.com/nathanieljohnston/QETLAB/blob/4f50dbf6ccd4a71c99d4f1afdcd85e9bfda5e9ac/IsSeparable.m#L203-L234).

In QETLAB, `p`'s shape is not pre-defined but I checked `p`'s shape at the end of the loop. The for loop indices also provide a hint of what's the shape of `p`.